backend/src/controller/plants.rs

```diff
@@ -1,25 +1,49 @@
 //! `Plants` endpoints.
 
-use crate::{config::db::Pool, service};
+use crate::{config::db::Pool, model::dto::PlantsSearchParameters, service};
 use actix_web::{
     get,
     web::{self, Data, Path},
     HttpResponse, Result,
 };
 
-/// Endpoint for fetching all [`PlantsDto`](crate::model::dto::PlantsDto).
+/// Endpoint for fetching or searching all [`PlantsSummaryDto`](crate::model::dto::PlantsSummaryDto).
+/// Search parameters are taken from the URLs query string (e.g. .../api/plants/search?query=example&limit=5).
+/// If no query parameters are provided, all plants are returned.
 ///
 /// # Errors
 /// * If the connection to the database could not be established.
 /// * If [web::block] fails.
 #[utoipa::path(
     context_path = "/api/plants",
     responses(
-        (status = 200, description = "Fetch all plants", body = Vec<PlantsDto>)
+        (status = 200, description = "Fetch or search for all plants by their common or species name", body = Vec<PlantsSummaryDto>)
     )
 )]
 #[get("")]
-pub async fn find_all(pool: Data<Pool>) -> Result<HttpResponse> {
+pub async fn find_all_or_search(
+    query: Option<web::Query<PlantsSearchParameters>>,
+    pool: Data<Pool>,
+) -> Result<HttpResponse> {
+    if let Some(query) = query {
+        search(query, pool).await
+    } else {
+        find_all(pool).await
+    }
```

Comment:
It might be cleaner if we don't call controllers from another controller.
You can directly go to the service layer like this.
```suggestion
    let response = match query {
        Some(parameters) => {
            web::block(move || service::plants::search(&pool, &parameters)).await??
        }
        None => web::block(move || service::plants::find_all(&pool)).await??,
    };
    Ok(HttpResponse::Ok().json(response))
```
You can also get rid of the other controllers.